cmd/job-executor-service-initcontainer/main.go

```diff
@@ -40,12 +56,54 @@ func main() {
 
 	fs := afero.NewOsFs()
 
-	var resourceHandler *api.ResourceHandler
-	if env.KeptnAPIToken != "" { // gets set as empty string from the job-executor-service if the env variable is not set
-		configurationServiceURL, _ := url.Parse(env.ConfigurationServiceURL)
-		resourceHandler = api.NewAuthenticatedResourceHandler(configurationServiceURL.String(), env.KeptnAPIToken, "x-token", nil, configurationServiceURL.Scheme)
+	baseURL, err := url.Parse(env.KeptnAPIURL)
+	if err != nil {
+		log.Fatalf("unable to parse the keptn api url: %s", err)
+	}
+
+	apiOptions := []func(*api.APISet){
+		api.WithScheme(baseURL.Scheme),
+	}
+
+	// Chose the authentication method from the give environment; This can either be token or oauth
+	if env.AuthMode == "token" {
+
+		// We only append a token when needing / having one, no token is needed
```

Comment:
What do you mean exactly with `no token is needed if the init-container lives in the same namespace as keptn` ? I thought authentication would be necessary anyways.... Is this a feature of the `api.APISet` ?

Reply:
It's a description that matches the old behavior, if there is no token we would use an unauthenticated resource handler to speak to the configuration service. I'm not sure if this is still possible with newer versions of Keptn and we maybe should just throw an error if there is no token specified ...